.github/workflows/regression-report.yml

```diff
@@ -73,30 +109,35 @@ jobs:
               content: 'rocket'
             })
 
-      - name: Checkout repository
-        uses: actions/checkout@v4
+      - name: Restore cached checkouts
+        uses: actions/cache@v4
+        with:
+          path: |
+            .
+            .ci-temp/checkstyle
+          key: ${{ runner.os }}-checkouts-${{ github.sha }}
 
       - name: Parse comment
         id: parse
         env:
           COMMENT_BODY: ${{ github.event.comment.body }}
         run: |
-          CONFIG_BUNDLE_PATH=$(echo "$COMMENT_BODY" | \
-            sed -E 's/.*(Github|GitHub), generate report for //g')
-          echo "CONFIG_BUNDLE_PATH=$CONFIG_BUNDLE_PATH"
-          ./.ci/append-to-github-output.sh "config_bundle_path" "$CONFIG_BUNDLE_PATH"
-
-      - name: Set config and projects links
-        id: set_links
-        run: |
-          BASE_URL="https://raw.githubusercontent.com/checkstyle/test-configs/main"
-          CONFIG_BUNDLE_PATH="${{ steps.parse.outputs.config_bundle_path }}"
-          CONFIG_LINK="$BASE_URL/$CONFIG_BUNDLE_PATH/config.xml"
-          PROJECTS_LINK="$BASE_URL/$CONFIG_BUNDLE_PATH/list-of-projects.properties"
-          echo "CONFIG_LINK=$CONFIG_LINK"
-          echo "PROJECTS_LINK=$PROJECTS_LINK"
-          ./.ci/append-to-github-output.sh "config_link" "$CONFIG_LINK"
-          ./.ci/append-to-github-output.sh "projects_link" "$PROJECTS_LINK"
+          if [[ "$COMMENT_BODY" =~ ^Git[Hh]ub,\ generate\ report\ for\ configs\ in\ PR\ \
+          description$ ]]; then
+            ./.ci/append-to-github-output.sh "mode" "configs_in_pr_description"
+          elif [[ "$COMMENT_BODY" =~ ^Git[Hh]ub,\ generate\ report\ for ]]; then
+            ./.ci/append-to-github-output.sh "mode" "use_config_bundle"
+            CONFIG_BUNDLE=$(echo "$COMMENT_BODY" | \
+              sed -E 's/^Git[Hh]ub, generate report for //g')
```

Comment:
this can be single line

Reply:
try this formatting to make easy to read:
```
        run: |
          if [[ "$COMMENT_BODY" =~ '^Git[Hh]ub, generate report for configs in PR description$' ]];
          then
            
            ./.ci/append-to-github-output.sh "mode" "configs_in_pr_description"
          
          elif [[ "$COMMENT_BODY" =~ ^Git[Hh]ub,\ generate\ report\ for ]]; then
            
            ./.ci/append-to-github-output.sh "mode" "use_config_bundle"
            CONFIG_BUNDLE=$(echo "$COMMENT_BODY" | sed -E 's/^Git[Hh]ub, generate report for //g')
            ./.ci/append-to-github-output.sh "config_bundle" "$CONFIG_BUNDLE"
          
          elif [[ "$COMMENT_BODY" =~ ^Git[Hh]ub,\ generate\ report\ by\ config\ from ]]; then
            
            ./.ci/append-to-github-output.sh "mode" "generate_config"
            INPUT_FILE=$(echo "$COMMENT_BODY" | sed -E 's/.*config from //g')
            ./.ci/append-to-github-output.sh "input_file" "$INPUT_FILE"
          
          else
            echo "No recognized pattern found in COMMENT_BODY."
            exit 1
          fi
```

Reply:
done